dkpro-jwpl-util/pom.xml

```diff
@@ -80,6 +75,15 @@
       <groupId>org.sweble.wikitext</groupId>
       <artifactId>swc-parser-lazy</artifactId>
     </dependency>
+    
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>commons-logging</groupId>
```

Comment:
Why do we need this dependency here? Can we avoid it?

Reply:
You mean commons-logging? I believe it is there because a fat-jar is built from this module and running that without any logging backend would be problematic.

If the module is also used as a dependency from other modules, then the dependency should be removed or changed in such a way that it gets included in the far JAR (which is build additionally to the normal jar?) but does not become a transitive dependency.

Reply:
Actually, compile-time vs. runtime does not really matter with respect to logging backends. What matters is transitivity. Both, compile and runtime scopes are transitive, so both are not really viable when this is used as a library by other modules.

Reply:
Thought / perspective was different:

Maven includes a dependency with _runtime_ scope in runtime and test classpaths, but not the _compile_ classpath.

The aim is to avoid JWPL code to reference `commons-logging` classes directly ("import by accident"). 

My comment is _not_ about JWPL being used from an external perspective ("as a library").

Reply:
In that case, I agree.

It would be nice if one could see from looking at a module whether it is meant to be a fat bundle or to be used by others as a library... I have no idea which module has which role in this project :(

Reply:
Agree

From what I know, all "...machine" modules are in fact tools (CLIs), that is, those are not meant to be used as a library.

Idea: We could refactor naming or module grouping, in a future release, 2.1 or maybe 3.x. This can be discussed deeper if we would open a discussion thread somewhere.

For now, I'm fine with having a modernized JWPL 2.0.0 release with room for future improvements or directions.